### PR TITLE
PPS: consume association cuts from DB

### PR DESCRIPTION
**Pull request by jan-kaspar**
#### PR description:

This PR updates the default PPS reconstruction configuration to consume the `PPSAssociationCuts` from DB. A small correction is made to the condition format to ensure that all class fields are correctly initialised.

This PR needs to be considered together with an updated GT. In particular it was tested with `121X_dataRun3_Candidate_2021_10_21_12_37_53`.

This is a technical PR, no changes in results are expected.

#### PR validation:

The plots below compare results before (blue) and after this PR (red dashed)
  * [dirsim_cmp.pdf](https://github.com/cms-sw/cmssw/files/7389343/dirsim_cmp.pdf): for PPS direct simulation (+ reconstruction), Run2 and Run3
  * [reco_cmp.pdf](https://github.com/cms-sw/cmssw/files/7389344/reco_cmp.pdf): for PPS reconstruction, Run2 data

No differences observed as expected.

### CODE REVIEW

**Comment by tvami** [comment]
Hi @jan-kaspar please note that code-checks have failed, please fix that so we can test this

**Comment by tvami** [comment]
test parameters:
  - pull_request  = https://github.com/cms-sw/cmssw/pull/35914

**Comment by tvami** [comment]
urgent

**Comment by francescobrivio** [comment]
@cmsbuild please test

**Comment by tvami** [comment]
@cms-sw/reconstruction-l2 sorry I've made a mistake by saying that https://github.com/cms-sw/cmssw/pull/35914 could go in independently of this... That PR is merged and now brakes the IB. Please move this PR higher in your priority list of reviewing, and sorry for my mistake

**Comment by slava77** [comment]
> @cms-sw/reconstruction-l2 sorry I've made a mistake by saying that #35914 could go in independently of this... That PR is merged and now brakes the IB. Please move this PR higher in your priority list of reviewing, and sorry for my mistake

reco code updates in this PR are almost trivial.
Still, the tests should pass and perhaps an alca signature should appear first.

**Comment by tvami** [comment]
+alca
 * the wfs 136.88811, 136.8311, 136.731 are now running fine
 * jenkins diffs are zero
 * unit test failure is known and has nothing to do with this PR

**Comment by tvami** [comment]
> > @cms-sw/reconstruction-l2 sorry I've made a mistake by saying that #35914 could go in independently of this... That PR is merged and now brakes the IB. Please move this PR higher in your priority list of reviewing, and sorry for my mistake
> 
> reco code updates in this PR are almost trivial. Still, the tests should pass and perhaps an alca signature should appear first.

Hi @slava77 the tests have converged, and AlCa has signed it (and I wrote privately to DQM)

**Comment by tvami** [comment]
> * unit test failure is known and has nothing to do with this PR

Just for the record: https://github.com/cms-sw/cmssw/issues/35868

**Comment by emanueleusai** [comment]
+1

**Comment by tvami** [comment]
Hi @cms-sw/reconstruction-l2 do you have any further comments?

**Comment by slava77** [comment]
+reconstruction

for https://github.com/cms-sw/cmssw/pull/35766	40e98d8
- code changes are in line with the PR description, changes in the reco code are essentially trivial, to remove non-DB ES specifications
- jenkins tests pass and comparisons with the baseline show no differences

**Comment by qliphy** [comment]
+1
to fix the IB issues (see discussions in https://github.com/cms-sw/cmssw/pull/35914)

**Comment by qliphy** [comment]
merge

File: CondFormats/PPSObjects/interface/PPSAssociationCuts.h [hunk 1]
```diff
@@ -48,8 +52,8 @@ class PPSAssociationCuts {
     std::vector<std::string> s_thresholds_;
 
     // TF1 representation of the cut parameters - for run time evaluations
-    std::vector<std::shared_ptr<TF1> > f_means_ COND_TRANSIENT;
-    std::vector<std::shared_ptr<TF1> > f_thresholds_ COND_TRANSIENT;
+    mutable std::vector<std::shared_ptr<TF1> > f_means_ COND_TRANSIENT;
```
Comment (Dr15Jones) [inline CondFormats/PPSObjects/interface/PPSAssociationCuts.h:55]: This is not thread safe and is likely causing the problems seen in the IB.

**Pull request merged**